resolver_balancer_ext_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ * Copyright 2014 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc_test
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/internal/balancer/stub"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+)
+
+// TestResolverBalancerInteraction tests:
+// 1. resolver.Builder.Build()
+// 2. resolver.ClientConn.UpdateState()
+// 3. balancer.Balancer.UpdateClientConnState()
+// 4. balancer.ClientConn.ResolveNow()
+// 5. resolver.Resolver.ResolveNow()
```

Comment:
Put some arrows so I know these operations are sync. Right now looks like independent operations, not within each other.

Reply:
Done